File: divisors_multiples1/cpp/no11653.cpp
#include <iostream>
#include <deque>

using namespace std;

int main() {
    int N;
    cin >> N;

    if (N == 1) {
        cout << endl;
    } else {
        deque<int> divisors;
        int x = 2;

        while (x * x <= N) {
            while (N % x == 0) {
                N /= x;
                divisors.push_back(x);
            }
            x += 1;
        }

        if (N > 1) {
            divisors.push_back(N);
        }

        for (int i = 0; i < divisors.size(); i++) {
            cout << divisors[i] << endl;
        }
    }
    return 0;
}

File: divisors_multiples1/python/no11653.py
N = int(input())

if N == 1:
    print()
else:
    divisors = []
    x = 2

    while x * x <= N:
        while N % x == 0:
            N //= x
            divisors.append(x)
        x += 1

    if N > 1:
        divisors.append(N)

    for i in range(len(divisors)):
        print(divisors[i])